README.md

```diff
@@ -29,28 +35,144 @@ $ git clone git@github.com:openfun/xblock-configurable-lti-consumer.git
+  - pattern: .*ltiapps\.net.*
+    fields:
+      launch_target: modal
+      custom_parameters: []
+      button_text: LTI demo
+      inline_height: 100
+      has_score: False
+      weight: 0
+      hide_launch: False
+      accept_grades_past_due: False
+      ask_to_send_username: True
+      ask_to_send_email: True
+  # A configuration that enforces a default behavior for all LTI consumer XBlocks
+  - pattern: .*
+    fields:
+      launch_target: in_window
+      ask_to_send_username: True
+      ask_to_send_email: True
+
+```
+
+The order of each configuration in this list is important because we will use, for a given
+XBlock, the first configuration that matchs its launch url.
+
+
+## Integration to Open edX Studio
+
+For now, this project requires a small fork of [edx/edx-platform](github.com/edx/edx-platform)
+if you want to add, for some of your configurations, preconfigured options in the
+`Advanced` component button of the Studio.
+
+In our opinion, this small fork is worth applying because:
+
+- it will save you from installing many XBlocks and help keep your Open edX installation manageable,
+- it allows you to automatically add a specific link for each LTI service you want to offer to your
+  instructors. They don't need to activate LTI and provide the credentials for each course... they
+  don't need to provide the launch url of the service for each XBlock they add...
+
+As of `open-release/hawthorn.1`, the patch to apply is:
+
+```diff
+diff --git a/cms/djangoapps/contentstore/views/component.py b/cms/djangoapps/contentstore/views/component.py
+index f65f281151..c8d0521c30 100644
+--- a/cms/djangoapps/contentstore/views/component.py
++++ b/cms/djangoapps/contentstore/views/component.py
+@@ -27,6 +27,15 @@ from xblock_django.models import XBlockStudioConfigurationFlag
+ from xmodule.modulestore.django import modulestore
+ from xmodule.modulestore.exceptions import ItemNotFoundError
+ 
++# Try to import the configurable LTI consumer xblock
++try:
++    from lti_consumer import add_dynamic_components
++except ImportError:
++    add_dynamic_components = None
++
++
+ __all__ = [
+     'container_handler',
+     'component_handler'
+@@ -359,6 +368,17 @@ def get_component_templates(courselike, library=False):
+         "support_legend": create_support_legend_dict()
+     }
+     advanced_component_types = _advanced_component_types(allow_unsupported)
++
++    # Read configuration setting to create specific buttons
++    if add_dynamic_components:
++        add_dynamic_components(
++            getattr(settings, "LTI_XBLOCK_CONFIGURATIONS", []),
++            advanced_component_templates,
++            categories,
++            create_template_dict,
++            course_advanced_keys
++        )
++
+     # Set component types according to course policy file
+     if isinstance(course_advanced_keys, list):
+         for category in course_advanced_keys:
 ```
```

Comment:
We can maybe store this patch as a gist an point to it in the documentation?

Reply:
so remove it from the documentation?

Reply:
Yes.

Reply:
Done